src/highlight/HighlightAnnotations.tsx

```diff
@@ -44,20 +44,34 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
         setActiveAnnotationId,
         setSelection,
         setMessage,
+        setMode,
+        setStaged,
+        setStatus,
         staged,
         status,
     } = props;
     const [highlightRef, setHighlightRef] = React.useState<HTMLAnchorElement | null>(null);
+    const [isPromoting, setIsPromoting] = React.useState<boolean>(false);
```

Comment:
I'm thinking to put this as a new `status` in `CreatorState`

Reply:
This state doesn't seem related to the Creator components, though. We have a separate promoter reducer for promoter-specific state. Can we open this as a separate PR, as well?